Bug 1193755 - Fixed url bar text squish during animation with defer workaround

sleroux opened the pull request:
Not exactly sure what is causing this inside UIKit but whenever we call becomeFirstResponder on a UITextField before or during an animation,
it makes the text squished at the beginning of the animation and stretches it out to normal size when the animation is finished. I've added a workaround by issuing the becomeFirstResponder call to be the next event on UIThread after the animation is processed.

thebnich commented on Client/Frontend/Browser/URLBarView.swift:
```diff
@@ -424,12 +424,10 @@ class URLBarView: UIView {
             // Clear any existing text, focus the field, then set the actual pasted text.
             // This avoids highlighting all of the text.
             locationTextField.text = ""
-            locationTextField.becomeFirstResponder()
```
Does `SearchTest` -> `testURLBarContextMenu` still pass with this change? I think the order here was important...if we become first responder after setting the text, it might highlight the whole string instead of the pasted portion.

sleroux replied:
Actually it looks like any tests that involve tapping the url bar by running:
```
        tester().tapViewWithAccessibilityIdentifier("url")
```
Seem to be failing on master at the moment. Would moving the if statement from above into this dispatch_async call work? I'll also see what's going on with these tests.

sleroux replied:
Wait maybe it's working now? I swear these tests start at random places in the test bundle. Still looking.

sleroux replied:
Ha - looks like testURLBarContextMenu fails in landscape but works in portrait...

sleroux commented:
@thebnich I've moved the pasting of the text into the async block and also moved the setting of the locationTextField to hidden inside common init instead of it being called all the time.

sleroux commented:
I'm going to go ahead and merge this in. The testURLContextMenu test seems to be passing now and the delayed/caching url text behavior is fixed :dancer:
